tests/integration/api_container_test.py

```diff
@@ -1226,6 +1251,29 @@ def test_attach_no_stream(self):
         output = self.client.attach(container, stream=False, logs=True)
         assert output == 'hello\n'.encode(encoding='ascii')
 
+    def test_attach_stream_and_cancel(self):
```

Comment:
This test can potentially cause some problems, seems flaky on Python 2.7 (fine on 3.5).